scripts/pre_commit_linter.py

```diff
@@ -580,6 +580,94 @@ def _check_bad_pattern_in_file(filename, content, pattern):
     return False
 
 
+def _check_bad_pattern_in_comments_docstrings(filename, content):
+    """Detects whether comments and docstrings end with a period.
+
+    Args:
+        filename: str. Name of the file.
+        content: str. Contents of the file.
+
+    Returns:
+        bool. True if there is bad pattern else false.
+    """
+    regexp_comment = r"#[^\n]*[^.,\n]\n( |\t)*(#(( |\t)*[A-Z])|[^#\s])"
+    message_comment = "There should be a period after comment end."
+    message_docstring = "There should be a period after docstring end."
+    punctuate = ['.',',',';',':']
+    excluded_words = ['utf','pylint:','http://']
+    excluded_dirs = ()
+    excluded_files = ()
+    if not (any(filename.startswith(excluded_dir)
+                for excluded_dir in excluded_dirs)
+            or filename in excluded_files):
+        bad_pattern_count = 0
+        # Finds comments with a missing period.
+        for matchobj in re.finditer(regexp_comment, content):
+            start = matchobj.start()+1
+            line_num = find_line(content, start)
```

Comment:
Try incorporating the functionality of `find_line` in the same loop instead of running another in a new function each time. This will increase the efficiency (Also, see if it is possible to use the already existing `_check_bad_pattern_in_file` function).

Reply:
Sure, I will incorporate the functionality in the same loop. It was not possible to use the function ``_check_bad_pattern_in_file`` since it checked for text line by line and in case of multiline comments we want to regex to find the pattern in the complete file. So, I used a new function to do that.